terraform-dev/terraform.tfvars

```diff
@@ -4,26 +4,29 @@ region          = "europe-west2"
 zone            = "europe-west2-b"
 location        = "EUROPE-WEST2"
 
-services = [
-  "storage.googleapis.com",
-  "iam.googleapis.com",
-  "appengine.googleapis.com",
-  "artifactregistry.googleapis.com",
-  "cloudscheduler.googleapis.com",
-  "bigquery.googleapis.com",
-  "bigquerydatatransfer.googleapis.com",
-  "run.googleapis.com",
-  "compute.googleapis.com",
-  "dns.googleapis.com",
-  "eventarc.googleapis.com",
-  "networkmanagement.googleapis.com",
-  "pubsub.googleapis.com",
-  "sourcerepo.googleapis.com",
-  "vpcaccess.googleapis.com",
-  "workflows.googleapis.com",
-  "iap.googleapis.com",
-  "secretmanager.googleapis.com",
-]
+services = concat(
+  [
+    "storage.googleapis.com",
+    "iam.googleapis.com",
+    "appengine.googleapis.com",
+    "artifactregistry.googleapis.com",
+    "cloudscheduler.googleapis.com",
+    "bigquery.googleapis.com",
+    "bigquerydatatransfer.googleapis.com",
+    "run.googleapis.com",
+    "compute.googleapis.com",
+    "dns.googleapis.com",
+    "eventarc.googleapis.com",
+    "networkmanagement.googleapis.com",
+    "pubsub.googleapis.com",
+    "sourcerepo.googleapis.com",
+    "vpcaccess.googleapis.com",
+    "workflows.googleapis.com",
+    "iap.googleapis.com",
+    "secretmanager.googleapis.com"
+  ],
+  var.enable_redis_session_store_instance ? ["redis.googleapis.com"] : []
```

Comment:
Neat, but unfortunately:

```
terraform apply
╷
│ Error: No value for required variable
│ 
│   on main-gcp.tf line 91:
│   91: variable "services" {
│ 
│ The root module input variable "services" is not set, and has no default value. Use a -var or -var-file command line argument to provide a value for this variable.
╵
╷
│ Error: Function calls not allowed
│ 
│   on terraform.tfvars line 7:
│    7: services = concat(
│    8:   [
│    9:     "storage.googleapis.com",
│   10:     "iam.googleapis.com",
│   11:     "appengine.googleapis.com",
│   12:     "artifactregistry.googleapis.com",
│   13:     "cloudscheduler.googleapis.com",
│   14:     "bigquery.googleapis.com",
│   15:     "bigquerydatatransfer.googleapis.com",
│   16:     "run.googleapis.com",
│   17:     "compute.googleapis.com",
│   18:     "dns.googleapis.com",
│   19:     "eventarc.googleapis.com",
│   20:     "networkmanagement.googleapis.com",
│   21:     "pubsub.googleapis.com",
│   22:     "sourcerepo.googleapis.com",
│   23:     "vpcaccess.googleapis.com",
│   24:     "workflows.googleapis.com",
│   25:     "iap.googleapis.com",
│   26:     "secretmanager.googleapis.com"
│   27:   ],
│   28:   var.enable_redis_session_store_instance ? ["redis.googleapis.com"] : []
│   29: )
│ 
│ Functions may not be called here.
```

Maybe do it in https://github.com/alphagov/govuk-knowledge-graph-gcp/blob/c80e6b2d1d7faa39d2f344cf69a9eddeed81000d/terraform/main-gcp.tf#L167 instead.  But there's no cost to enabling an API that isn't used, so I think this might be overkill.

Reply:
Thank you!